src/MemoryRouter.tsx

```diff
@@ -132,6 +147,84 @@ export class MemoryRouter extends BaseRouter {
     return this._setCurrentUrl(url, as, options, "replace");
   };
 
+  /**
+   * Sets the current MemoryHistory.
+   */
+  public setCurrentHistory = (history: MemoryHistory) => {
+    this._history = history;
+    this.setCurrentUrl(history.location.pathname + history.location.search + history.location.hash);
+  };
+
+  /**
+   * Returns the current MemoryHistory.
+   * history.state property represents the previous location's state in MemoryHistory.
+   */
+  get history() {
+    return {
+      ...this._history,
+      createHref(to) {
+        throw new Error("You cannot use history.createHref() because it is a stateless.");
+      },
+      push(to, state) {
+        throw new Error("You cannot use history.push() because it is a stateless.");
```

Comment:
refer to react-router to return stateless history.

https://github.com/remix-run/react-router/blob/9f36087045fd8577ed1aa23f119e5b9e83933ab4/packages/react-router-dom-v5-compat/lib/components.tsx#L108-L114